jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java

```diff
@@ -350,6 +350,60 @@ public void clearEventListeners()
         _sessionListeners.clear();
         _sessionIdListeners.clear();
     }
+    
+    
+    /**
+     * Call the session lifecycle listeners
+     * @param session the session on which to call the lifecycle listeners
+     */
+    protected void callSessionDestroyedListeners (Session session)
+    {
+        if (session == null)
+            return;
+        
+        if (_sessionListeners!=null)
+        {
+            HttpSessionEvent event=new HttpSessionEvent(session);      
+            for (int i = _sessionListeners.size()-1; i>=0; i--)
+            {
+                _sessionListeners.get(i).sessionDestroyed(event);
+            }
+        }
+    }
+    
+    /**
+     * Call the session lifecycle listeners
+     * @param session the session on which to call the lifecycle listeners
+     */
+    protected void callSessionCreatedListeners (Session session)
+    {
+        if (session == null)
+            return;
+        
+        if (_sessionListeners!=null)
+        {
+            HttpSessionEvent event=new HttpSessionEvent(session);      
+            for (int i = _sessionListeners.size()-1; i>=0; i--)
+            {
+                _sessionListeners.get(i).sessionCreated(event);
+            }
+        }
+    }
+    
+    
+    protected void callSessionIdListeners (Session session, String oldId)
+    {
+        //inform the listeners
+        if (!_sessionIdListeners.isEmpty())
+        {
+            HttpSessionEvent event = new HttpSessionEvent(session);
+            for (HttpSessionIdListener l:_sessionIdListeners)
+            {
+                l.sessionIdChanged(event, oldId);
```

Comment:
If more then 1 listener, and the first one fails, the others will not be called.

Reply:
see above